core/src/main/java/org/mule/transport/polling/PollingMessageSource.java

```diff
@@ -0,0 +1,421 @@
+            public void run() throws Exception
+            {
+                PollingMessageSource.this.performPoll();
+            }
+
+            @Override
+            public void stop() throws MuleException
+            {
+                PollingMessageSource.this.stop();
+            }
+        }, muleContext.getExceptionListener());
+    }
+
+    public long getFrequency()
+    {
+        return frequency;
+    }
+
+    public void setFrequency(long value)
+    {
+        if (value <= 0)
+        {
+            frequency = DEFAULT_POLL_FREQUENCY;
+        }
+        else
+        {
+            frequency = value;
+        }
+    }
+
+    public TimeUnit getTimeUnit()
+    {
+        return timeUnit;
+    }
+
+    public void setTimeUnit(TimeUnit timeUnit)
+    {
+        this.timeUnit = timeUnit;
+    }
+
+    /**
+     * Check whether polling should take place on this instance.
+     */
+    public final void performPoll() throws Exception
+    {
+        if (!pollOnPrimaryInstanceOnly() || flowConstruct.getMuleContext().isPrimaryPollingInstance())
+        {
+            poll();
+        }
+    }
+
+    private boolean pollOnPrimaryInstanceOnly()
+    {
+        return true;
+    }
+
+    /**
+     * <p>
+     * Helper method to create {@link org.mule.api.schedule.Scheduler} names
+     * </p>
+     */
+    private String schedulerNameOf(FlowConstruct flowConstruct)
+    {
+        return String.format(POLLING_SCHEDULER_NAME_FORMAT, flowConstruct.getName(), this.hashCode());
+    }
+
+    /**
+     * <p>
+     * The poll message source, configured inside the poll element in the xml configuration. i.e.:
+     * <pre>
+     * {@code
+     * <poll>
+     *       <sfdc:query query=""/>
+     * </poll>
+     * </pre>
+     * </p>
+     */
+    protected MessageProcessor sourceMessageProcessor;
+
+    /**
+     * <p>
+     * The {@link MessageProcessorPollingOverride} that affects the routing of the {@link MuleEvent}
+     * </p>
+     */
+    protected MessageProcessorPollingOverride override;
+
+    public void poll() throws Exception
+    {
+        MuleMessage request = new DefaultMuleMessage(StringUtils.EMPTY, (Map<String, Object>) null, muleContext);
+        pollWith(request);
+    }
+
+    public void pollWith(final MuleMessage request) throws Exception
+    {
+        ExecutionTemplate<MuleEvent> executionTemplate = TransactionalErrorHandlingExecutionTemplate.createMainExecutionTemplate(muleContext, flowConstruct.getExceptionListener());
+        try
+        {
+            final MessageProcessorPollingInterceptor interceptor = override.interceptor();
+            MuleEvent muleEvent = executionTemplate.execute(new ExecutionCallback<MuleEvent>()
+            {
+                @Override
+                public MuleEvent process() throws Exception
+                {
+                    //TODO review what to put as exchange pattern. It used to be the one from the inbound endpoint if the message processor was an IE
```

Comment:
Should be one-way. If this is going to be pending, then add a reference to the jira